src/app/pages/SearchResultsPage/SearchResultsList.tsx

```diff
@@ -0,0 +1,56 @@
+import { FC } from 'react'
+import { Network } from '../../../types/network'
+import { useTranslation } from 'react-i18next'
+import { ResultsGroup } from './ResultsGroup'
+import { BlockDetailView } from '../BlockDetailPage'
+import { RouteUtils } from '../../utils/route-utils'
+import { TransactionDetailView } from '../TransactionDetailPage'
+import { AccountDetailsView } from '../AccountDetailsPage'
+import { SearchQueries } from './hooks'
+
+export const SearchResultsList: FC<{
+  network: Network
+  searchQueries: SearchQueries
+  roseFiatValue: number | undefined
+}> = ({ network, searchQueries, roseFiatValue }) => {
+  const { t } = useTranslation()
+  return (
+    <>
+      <ResultsGroup
+        title={t('search.results.blocks.title')}
+        results={searchQueries.blockHeight.results.filter(result => result.network === network)}
```

Comment:
Eww

Reply:
Would you please elaborate? Is this about performance, and code readability?

Reply:
We have one implementation in a higher layer for counting results, and completely different implementation in this layer for showing the results

Reply:
*different implementation of counting/filtering all results

Reply:
Counting and filtering logic was unified in https://github.com/oasisprotocol/explorer/pull/449/commits/050a9557f6966937a9fbd798b14c9ff2755280c0


https://github.com/oasisprotocol/explorer/pull/449/commits/7e9666708e8275b2fd724a0f20c34f2796306a6b and https://github.com/oasisprotocol/explorer/pull/449/commits/f060f9cb9d901968c16a19aae75b4c754a4867e4 refactored from passing in `all results + filter` to passing `filtered results`.

:tada: very readable